pydoctor/epydoc2stan.py

```diff
@@ -552,8 +561,93 @@ def __init__(self, annotation: ast.expr) -> None:
         self.annotation = annotation
 
     def to_stan(self, docstring_linker: DocstringLinker) -> Tag:
-        src = astor.to_source(self.annotation).strip()
-        ret: Tag = tags.code(src)
+        tag: Tag = tags.code
+        tag(_AnnotationFormatter(docstring_linker).visit(self.annotation))
+        return tag
+
+
+class _AnnotationFormatter(ast.NodeVisitor):
+
+    def __init__(self, linker: DocstringLinker):
+        super().__init__()
+        self.linker = linker
+
+    def _handle_name(self, identifier: str) -> Tag:
+        url = self.linker.resolve_identifier(identifier)
+
+        tag: Tag
+        if url is None:
+            tag = tags.transparent(identifier)
+        else:
+            tag = tags.a(identifier, href=url, class_='code')
```

Comment:
it looks like there is no coverage for this line.

Reply:
Yes, that is also a consequence of the `RecordingAnnotationLinker` test helper always returning "not found".

I don't want to deal with HTML when comparing the test results though, at that is messy and brittle. And just executing the line without comparing its output, to check that it doesn't crash, is not very valuable now that we have type checking, since mypy will catch the majority of issues that you can find this way. It might be useful in some cases, for example when there are non-trivial asserts in the code, but I don't think there are any on this code path.